tests/infill/SierpinskiFillTest.cpp

```diff
@@ -0,0 +1,84 @@
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/slight_gradient.png", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/simple.png", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/gray.png", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/triangle.png", aabb);
+    
+    DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/hitler.jpg", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/cheshire_cat.jpg", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/vader.jpg", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/sinterklaas.jpg", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/diamond.jpg", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/trex.jpeg", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/soulpilot.jpg", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/soulpilot_dark.jpg", aabb);
+//     DensityProvider* subdivider = new ImageBasedDensityProvider("/home/t.kuipers/Documents/PhD/Cross Fractal/cura-logo.jpg", aabb);
+//     subdivider = new UniformSubdivider();
+//     srand(1);
+    int max_depth = 1;
+    for (coord_t size = line_width; size < aabb.max.X; size = size << 1)
+    {
+        max_depth += 2;
+    }
+    bool dithering = true;
+    {
+        SierpinskiFill f(*subdivider, aabb, max_depth, line_width, dithering);
+        SVG::Color color = SVG::Color::GREEN;
+        switch (max_depth % 4)
+        {
+            default:
+            case 0: color = SVG::Color::GREEN; break;
+            case 1: color = SVG::Color::RED; break;
+            case 2: color = SVG::Color::BLUE; break;
+            case 3: color = SVG::Color::YELLOW; break;
+        }
+        color = SVG::Color::RAINBOW;
+        color = SVG::Color::BLACK;
+//         svg.writePolygon(f.generateCross(1200, 16), color, 4);
+//         svg.writePolygon(f.generateCross(), color, drawing_line_width);
+        svg.writeAreas(f.generateCross(), SVG::Color::WHITE, color, drawing_line_width);
+//         svg.writePoints(f.generateCross());
+//         svg.writePolygon(f.generateSierpinski(), color);
+//         f.debugOutput(svg);
+    }
+    
+}
```

Comment:
Aww I was hoping for an actual unit test :( This just generates an SVG file.

Better not register it as a unit test then. In fact, I think it's better to remove this class altogether.

Reply:
It's still a nice test, but it's not a unit test....

I'll remove it along with the actual Sierpinski curve fucntion.